articles/getting-started-with-matlab/README.md

```diff
@@ -0,0 +1,207 @@
+# Getting-started-with-MATLAB
+### Introduction
+MATLAB (an abbreviation of "matrix laboratory") is a programming language and numerical computing environment developed by MathWorks.
+
+### Prerequisites
+- Good understanding of any programming language.
+- MATLAB installed, you can download it from [here](https://www.mathworks.com/downloads/).
+
+### Uses of MATLAB
+- Signal Processing and Communications.
+- Data analysis and visualization.
+- Scientific and engineering graphics.
+- Developing algorithms.
+- Modeling, simulation, and prototyping.
+
+MATLAB has an additional package "Simulink" which you can use for graphical multi-domain simulation and model-based design for dynamic and embedded systems. And you can get access to the symbolic computing abilities by using an optional toolbox that uses the MuPAD symbolic engine.
+
+### Lets code!
+In heart, MATLAB is a calculator. For example, try: `1+1`, `sqrt(4)`, or `sin(pi/2)`.
+
+Also, you can present complex numbers by:
+| Symbol  | Function  |   
+|---|---|
+| abs(x)  | absolute value.  |   
+| angle  | angle of complex number x.  |   
+| conj(x)  | complex conjugate of x.  |   
+| imag(x)  | imaginary part of complex number x.  |   
+| real(x)  | real part of complex number x . |   
+
+#### Vector
+The vector is a one-dimensional array of numbers.
+
+##### Row vector
+You can create row vectors by:
+```matlab
+a = [ 1 2 3 4]
+```
+Or by using a comma to delimit the elements. MATLAB will return the following result:
+```matlab
+a =
+
+     1     2     3     4
+```
+##### Column vector
+You need to use semicolons between values. For example:
+```matlab
+a = [1;2;3;4]
+```
+And the output will be:
+```matlab
+a =
+
+     1
+     2
+     3
+     4
+```
+If you want to output a specific range of elements, you have to reference the vector with a colon. For example:
+```matlab
+values=[6 513 21 9 23 1 0];
+values(2:5)
+```
+The result will be:
+```matlab
+ans =
+
+   513    21     9    23
+```
+And if you want to list all of the vector elements, you can use `values(:)`.
+
+#### Matrix
+The matrix is a two-dimensional array of numbers.
+
+To create a matrix, you have to enter the elements in each row with space or colon between them. Then you have to use semicolons to delimit the rows as we saw above. For example:
+```matlab
+values=[1 2 3 4;5 6 7 8; 9 10 11 12]
+```
+The result:
+```matlab
+values =
+
+     1     2     3     4
+     5     6     7     8
+     9    10    11    12
+```
+Now let's learn how to create a smaller matrix from our existing one. For example, if you execute `values(:, 1:2)` the result will be:
```

Comment:
Explain the slicing operation in detail. What is the comma doing? Why is the first value left blank? Explain these in detail. These concepts explained in-depth add tremendous value and differentiate the content from the existing ones,